src/caustics/light/point_source.py

```diff
@@ -0,0 +1,163 @@
+# mypy: disable-error-code="operator,union-attr"
+from typing import Optional, Union, Annotated
+
+from torch import Tensor
+
+from .base import Source, NameType
+from ..parametrized import unpack
+from ..packed import Packed
+from . import func
+
+__all__ = ("PointSource",)
+
+
+class PointSource(Source):
```

Comment:
Point source might not be the right name for this class. A point source should have a delta function for its brightness, which isn't very practical. Perhaps a more direct name for this class like `StarSource` since that is what it is representing and it includes limb darkening?

Reply:
Good call. I was originally planning to implement a point source and then add a finite extent, but this isn't realistic. I renamed all relevant classes/functions to use "star" instead of "point"